pkg/buildpack/builder.go

```diff
@@ -268,15 +268,14 @@ func validateBuildpacks(mainBP Buildpack, depBPs []Buildpack) error {
 		bpd := bp.Descriptor()
 		for _, orderEntry := range bpd.Order {
 			for _, groupEntry := range orderEntry.Group {
-				if _, ok := depsWithRefs[groupEntry.BuildpackInfo.FullName()]; !ok {
+				if _, ok := depsWithRefs[groupEntry.BuildpackInfo.FullNameWithVersion()]; !ok {
 					return errors.Errorf(
-						"buildpack %s references buildpack %s which is not present",
-						style.Symbol(bpd.Info.FullName()),
-						style.Symbol(groupEntry.FullName()),
+						"Invalid buildpack %s defined in order group.",
+						style.Symbol(groupEntry.FullNameWithVersion()),
 					)
```

Comment:
I feel like the new error message is a little too opaque. If the version was missing, all we would see is:

```
Invalid buildpack 'bp.missing.id@' defined in order group.
```


Based on our conversation [here](https://github.com/buildpacks/pack/issues/1384#issuecomment-1116129061), what if we had two errors:

1. the existing error for when a buildpack is genuinely missing
2. a new error for when the referenced buildpack is missing a version:
```
buildpack %s must specify a version when referencing buildpack %s.
```

Reply:
Updated the error as discussed, added test as well.
Let me know if more work is required.